modules/adspendBidAdapter.js

```diff
@@ -0,0 +1,150 @@
+import * as utils from 'src/utils';
+import { config } from 'src/config';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'adspend';
+const BID_URL = '//rtb.com.ru/headerbidding-bid';
+const SYNC_URL = '//rtb.com.ru/headerbidding-sync?uid={UUID}';
+const COOKIE_NAME = 'hb-adspend-id';
+const UUID_LEN = 36;
+const COOKIE_EXPIRE = 600;
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['adspend'],
+  supportedMediaTypes: [BANNER],
+
+  onBidWon: function(winObj) {
+    const requestId = winObj.requestId;
+    const cpm = winObj.cpm;
+    const event = getWinEvent(requestId).replace(/\$\{AUCTION_PRICE\}/, cpm);
+
+    fetch(event);
+  },
+
+  isBidRequestValid: function(bid) {
+    const conf = config.getConfig();
+    return !!(conf.currency &&
+      conf.currency.adServerCurrency &&
+      bid.crumbs.pubcid &&
+      utils.checkCookieSupport() &&
+      utils.cookiesAreEnabled()
+    );
+  },
+
+  buildRequests: function(bidRequests, bidderRequest) {
+    const req = bidRequests[Math.floor(Math.random() * bidRequests.length)];
+
+    const conf = config.getConfig();
+    const cur = conf.currency.adServerCurrency;
+
+    const bidfloor = req.params.bidfloor;
+
+    const payload = {
+      'id': req.bidId,
+      'site': {
+        'id': req.crumbs.pubcid,
+        'domain': document.domain
+      },
+      'device': {
+        'ua': navigator.userAgent,
+        'ip': ''
+      },
+      'user': { 'id': getUserID() },
+      'imp': [
+        {
+          'id': req.params.placement,
+          'tagId': req.params.tagId,
+          'banner': { 'format': getFormats(req.sizes) },
+          'bidfloor': bidfloor !== undefined ? Number(bidfloor) : 1,
+          'bidfloorcur': cur,
+          'secure': 0
+        }
+      ],
+      'cur': [
+        cur
+      ],
+      'tmax': bidderRequest.timeout
+    };
+
+    return {
+      method: 'POST',
+      url: BID_URL,
+      data: JSON.stringify(payload),
+    };
+  },
+
+  interpretResponse: function(resp, {bidderRequest}) {
+    const bids = [];
+
+    if (resp.body === '') return bids;
+
+    const respBid = resp.body.seatbid[0].bid[0];
+    const requestId = resp.body.id;
+
+    document.cookie =
+      `on_win_event__${requestId}=${respBid.nurl}; max-age=${COOKIE_EXPIRE}`;
```

Comment:
I see you using this cookie in the `getWinEvent`.  Is there something that necessitates this being stored in a cookie rather than just in memory?

Reply:
No. Looks like you're right, I should store it in-memory